src/utils/walletConnect.js

```diff
@@ -141,18 +144,19 @@ export const mapCallRequestToTransactionPayload = (
     to = methodToAddress;
   }
 
-  const ethAssetData = getAssetData(accountAssets, supportedAssets, ETH);
+  const { chainId } = callRequest;
+  const chain = chainFromChainId[chainId];
+  const chainNativeSymbol = nativeAssetSymbolPerChain[chain];
+  const chainNativeDecimals = nativeAssetDecimalsPerChain[chain];
```

Comment:
Maybe we could get `getNativeAsset(chain)` that would return `Asset` or `AssetData` or `AssetCore` instead of these separated arrays, wdyt?

Reply:
Not sure since we have `AssetData` as type and this method might confuse in further development.